tests/element_test_vectors.rs

```diff
@@ -198,30 +178,29 @@ trait ElementApi {
     ///
     /// This would have two groups, one with direct values that will be compared and another
     /// with embedded Ion text that will be parsed and compared.
-    fn read_group<F1, F2>(file_name: &str, value_assert: F1, group_assert: F2) -> IonResult<()>
+    fn read_group<F>(file_name: &str, value_assert: F) -> IonResult<()>
     where
-        // group index, value 1 index, value 1, value 2 index, value 2
-        F1: Fn(usize, String, &Element, String, &Element),
-        F2: Fn(&Vec<Element>, &Vec<Element>),
+        // group index, value 1 name/index, value 1, value 2 name/index, value 2
+        F: Fn(usize, String, &Element, String, &Element),
     {
         let group_lists = Self::read_file(file_name)?;
         for (group_index, group_container) in group_lists.iter().enumerate() {
-            // every grouping set is a list/sexp
+            // every grouping set is a list/sexp/struct
             // look for the embedded annotation to parse/test as the underlying value
             let is_embedded = group_container.annotations().contains("embedded_documents");
-            match (group_container.value(), is_embedded) {
-                (Value::List(group), true) | (Value::SExp(group), true) => {
-                    Self::read_group_embedded(group, &group_assert)?;
-                }
-                (Value::Struct(group), true) => {
-                    let sequence: Sequence = group
-                        .to_owned()
-                        .fields()
-                        .map(|(_, v)| v.to_owned())
-                        .collect();
-                    Self::read_group_embedded(&sequence, &group_assert)?;
-                }
-                (Value::List(group), false) | (Value::SExp(group), false) => {
+            match group_container.value().to_owned() {
```

Comment:
Is the `to_owned()` needed? It looks like we're just iterating over whatever we get.

Reply:
A `to_owned()` is necessary somewhere because I can't have the `if` expression returning `Sequence` from one branch and `&Sequence` from another, but I was able to move the `to_owned()` call so that it is used as late as possible.